pkg/controller/ec2/flowlog/setup.go

```diff
@@ -0,0 +1,313 @@
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithInitializers(),
+			managed.WithPollInterval(o.PollInterval),
+			managed.WithLogger(o.Logger.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
+			managed.WithConnectionPublishers(cps...)))
+
+}
+
+func determineResourceIdsAndType(cr *svcapitypes.FlowLog) ([]*string, *string) {
+	vpcResourceType := svcsdk.FlowLogsResourceTypeVpc
+	transitGatewayResourceType := "TransitGateway"
+	transitGatewayAttachmentResourceType := "TransitGatewayAttachment"
+	subnetResourceType := svcsdk.FlowLogsResourceTypeSubnet
+	networkInterfaceResourceType := svcsdk.FlowLogsResourceTypeNetworkInterface
+	if cr.Spec.ForProvider.VPCID != nil {
+		return []*string{cr.Spec.ForProvider.VPCID}, &vpcResourceType
+	}
+
+	if cr.Spec.ForProvider.TransitGatewayID != nil {
+		return []*string{cr.Spec.ForProvider.TransitGatewayID}, &transitGatewayResourceType
+	}
+
+	if cr.Spec.ForProvider.TransitGatewayAttachmentID != nil {
+		return []*string{cr.Spec.ForProvider.TransitGatewayAttachmentID}, &transitGatewayAttachmentResourceType
+	}
+
+	if cr.Spec.ForProvider.SubnetID != nil {
+		return []*string{cr.Spec.ForProvider.SubnetID}, &subnetResourceType
+	}
+
+	if cr.Spec.ForProvider.NetworkInterfaceID != nil {
+		return []*string{cr.Spec.ForProvider.NetworkInterfaceID}, &networkInterfaceResourceType
+	}
+
+	return nil, nil
+}
+
+func preObserve(ctx context.Context, cr *svcapitypes.FlowLog, obj *svcsdk.DescribeFlowLogsInput) error {
+	externalName := meta.GetExternalName(cr)
+	obj.FlowLogIds = []*string{&externalName}
+	return nil
+}
+
+func filterList(cr *svcapitypes.FlowLog, list *svcsdk.DescribeFlowLogsOutput) *svcsdk.DescribeFlowLogsOutput {
+	if len(list.FlowLogs) == 0 {
+		return list
+	}
+	flowLogs := []*svcsdk.FlowLog{}
+	for _, f := range list.FlowLogs {
+		if aws.StringValue(f.FlowLogId) == meta.GetExternalName(cr) {
+			flowLogs = append(flowLogs, f)
+		}
+	}
+	list.FlowLogs = flowLogs
+	return list
+}
+
+func postObserve(ctx context.Context, cr *svcapitypes.FlowLog, obj *svcsdk.DescribeFlowLogsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	cr.SetConditions(xpv1.Available())
+	return obs, err
+}
+
+func preCreate(_ context.Context, cr *svcapitypes.FlowLog, obj *svcsdk.CreateFlowLogsInput) error {
+
+	if cr.Spec.ForProvider.S3BucketLogDestination != nil {
+		obj.LogDestination = cr.Spec.ForProvider.S3BucketLogDestination
+		if cr.Spec.ForProvider.S3BucketSubfolder != nil {
+			// If a subfolder is given, we append it to the ARN managed by crossplane
+			destination := *obj.LogDestination + "/" + *cr.Spec.ForProvider.S3BucketSubfolder + "/"
+			obj.LogDestination = &destination
+		}
+	}
+
+	if cr.Spec.ForProvider.CloudWatchLogDestination != nil {
+		obj.LogDestination = cr.Spec.ForProvider.CloudWatchLogDestination
+	}
+
+	if cr.Spec.ForProvider.Tags != nil {
```

Comment:
Think we have in other resources the tagger stuff in extra function ?

Reply:
Moved generation of tags in new function generateTagSpecifications